### DISCUSSION TITLE
Integration with TopicLongTable

### DISCUSSION THREAD

**Discussion by billdenney** [davidgohel/flextable#414] [Ideas]
Is there interest in integrating topiclongtable with flextable?  (I don't know of an html or word equivalent tool.)

I recently learned of flextable.  While looking through it, I noticed that it has some functionality that is similar to the topiclongtable TeX package:  https://ctan.org/pkg/topiclongtable?lang=en

What topiclongtable does is automatically groups together repeated rows in a table (see the figure on page 2 of the pdf documentation in the pdf documentation on CTAN above).  But, more important is that is solves a problem when tables span multiple pages.

When a table spans multiple pages, the default would be that you don't know what the rowspanned column is.  With topiclongtable, it automatically repeats the value in the column when going to the next page.  A visual example (from pages 3-4 of the pdf documentation on ctan) is:

Page 3:
![image](https://user-images.githubusercontent.com/10359931/175403754-72fcbed6-73da-46eb-9b5d-1f502e1f4ab3.png)

Page 4:
![image](https://user-images.githubusercontent.com/10359931/175403862-d92d86d6-0dd1-4294-90a1-d38946181b79.png)

I've also written an R interface to the CTAN package here: https://github.com/billdenney/TopicLongTableR

**Comment by davidgohel** [comment]
Hello 

Yes, it's interesting. Some users asked about something like that: https://github.com/davidgohel/flextable/issues/394

Is it possible to use have to feature that display "(Cont.)" but not use the other feature? (I am not a latex expert, sorry if the question is stupid)

**Reply by billdenney** [reply]
The request in #394 is a feature of the standard long table package.  That would also be possible with topiclongtable, but the real difference with topiclongtable is that it groups together values in the same column as a long rowspan and if there is a page break, it shows you the value again.

In the example above, where "(cont)" is in the cell, otherwise the value would be blank.

**Reply by davidgohel** [reply]
OK, the API provided by flextable is agnostic of formats. The calculation of horizontal merging happens in flextable before printing, not when rendering in PDF or HTML/Word/PPTX. 

Is it possible to let me merge some cells and not others cells. If successive values are A, A, A, A, we must merge the four, right?

**Reply by billdenney** [reply]
I don't know if it would work to merge some rows and not merge others in the same column.  That isn't the functionality that was intended at the TeX package design, but that doesn't mean it wouldn't work.

But, the package was designed to merge as rowspans only when there are no changes at the current cell or any cells to the left.  (This concept is clearer by looking at the examples in the CTAN PDF documentation.)

If full rowspan merge control is the goal, then this isn't the right solution.  If clarity of what was merged across pages is the goal, then this is the right solution (in my opinion).

**Reply by davidgohel** [reply]
The final result is indeed very interesting. I will try in the next development run
